src/Contract/Test/Helpers.purs

```diff
@@ -0,0 +1,335 @@
+{- | This module is intended to be used for running custom E2E-tests -}
```

Comment:
> However, I am not sure about the CLI parser. How reusable it is? I think it does provide some value in case the users want to make their own binary that works exactly the same.

I feel the same about it. I tend to leave it exported, because it might be useful.